src/functionaltests/java/com/ericsson/ei/notifications/ttl/TestTTLSteps.java

```diff
@@ -0,0 +1,225 @@
+            assertEquals("Failed to create aggregated object in database",true, isInserted);
+
+            List<String> result = mongoDBHandler.getAllDocuments(aggregationDataBaseName,
+                    aggregationCollectionName);
+            assertEquals("Database " + aggregationDataBaseName + " is not empty.", false, result.isEmpty());
+        }
+
+        @When("^I sleep for \"([^\"]*)\" seconds")
+        public void i_sleep_for_time(int sleepTime) throws Throwable {
+            LOGGER.debug("Sleeping for " + sleepTime + " at " + LocalTime.now());
+            // The background task deleting notifications from Mongo DB
+            // runs every 60 seconds
+            TimeUnit.SECONDS.sleep(sleepTime);
+        }
+
+        @Then("^\"([^\"]*)\" has been deleted from \"([A-Za-z0-9_]+)\" database$")
+        public void has_been_deleted_from_database(String collection, String database) {
+            LOGGER.debug("Checking " + collection + " in " + database);
+            List<String> result = mongoDBHandler.getAllDocuments(database, collection);
+            assertEquals("Database is not empty.", true, result.isEmpty());
+        }
+
+        // SCENARIO @TestNotificationRetries
+
+        @Given("^Subscription is created$")
+        public void create_subscription_object() throws IOException, JSONException {
+            LOGGER.debug("Starting scenario @TestNotificationRetries.");
+            mongoDBHandler.dropCollection(missedNotificationDatabase, missedNotificationCollectionName);
+
+            String subscriptionStr = FileUtils.readFileToString(new File(SUBSCRIPTION_FILE_PATH), "utf-8");
+
+            // replace with port of running mock server
+            subscriptionStr = subscriptionStr.replaceAll("\\{port\\}",
+                    String.valueOf(clientAndServer.getPort()));
+
+            subscriptionObject = new ObjectMapper().readTree(subscriptionStr);
+            assertEquals(false, subscriptionObject.get("notificationMeta").toString().contains("{port}"));
+        }
+
+        @When("^I fail to inform subscriber$")
+        public void trigger_notification() throws IOException {
+            JsonNode aggregatedObject = getJSONFromFile(AGGREGATED_OBJECT_FILE_PATH);
+            informSubscription.informSubscriber(aggregatedObject.toString(), subscriptionObject);
+        }
+
+        @Then("^Verify that request has been retried")
+        public void verify_request_has_been_made() throws JSONException {
+            String retrievedRequests = mockServerClient.retrieveRecordedRequests(request()
+                    .withPath(ENDPOINT), Format.JSON);
+            JSONArray requests = new JSONArray(retrievedRequests);
+
+            // received requests include number of retries
+            assertEquals(2, requests.length());
+        }
+
+        @Then("^Missed notification is in database$")
+        public void missed_notification_is_in_database() {
+            List<String> result = mongoDBHandler.getAllDocuments(
```

Comment:
If you know the subscriptionName or some other parameters from the object, its better to query on that parameter when checking the MongoDb. Then the check is more properly done.